lib/crystalball/case_map.rb

```diff
@@ -3,17 +3,21 @@
 module Crystalball
   # Data object to store execution map for specific example
   class CaseMap
-    attr_reader :uid, :file_path, :affected_files
+    attr_reader :example, :affected_files
     extend Forwardable
 
     delegate %i[push] => :affected_files
+    delegate %i[file_path] => :example
 
     # @param [String] id of example
     # @param [Array<String>] list of files affected by example
     def initialize(example, affected_files = [])
-      @uid = example.id
-      @file_path = example.file_path
+      @example = example
```

Comment:
![No!](https://media.giphy.com/media/WO5GDSsx76rTy/giphy.gif)
That saves the reference of an example => never GC.

Reply:
Let's keep simple things simple. two strings are way better than one unknown object.